string client_ID
time stamp
int64 sequence_number